fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/FedoraResourceImpl.java

```diff
@@ -257,7 +260,22 @@ public void delete() {
             final Iterator<Property> inboundProperties = Iterators.concat(references, weakReferences);
 
             while (inboundProperties.hasNext()) {
-                inboundProperties.next().remove();
+                final Property prop = inboundProperties.next();
+                final List<Value> newVals = new ArrayList<Value>();
+                final Iterator<Value> propIt = JcrPropertyFunctions.property2values.apply(prop);
+                while (propIt.hasNext()) {
```

Comment:
Oh, shoot, you're right. You can't. But you can still do better than this via [Guava's `removeIf`](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/collect/Iterators.html#removeIf%28java.util.Iterator,%20com.google.common.base.Predicate%29).